### PR TITLE
enhancement for grid layout

### PR DESCRIPTION
**Pull request by imanjra**
## What changes are proposed in this pull request?

Several enhancements for GridView to better organize content

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
  - Introduced `DimensionRefresher` component to enhance panel dimension handling.

- **Improvements**
  - Enhanced grid layout logic in the GridView component.
  - Added scrollable content support in panels.
  - Improved dimension calculation with throttling and separate width/height handling.

- **Bug Fixes**
  - Corrected grid direction logic for horizontal layouts.

- **Refactor**
  - Reorganized imports for better code maintainability.

### CODE REVIEW

File: app/packages/core/src/plugins/SchemaIO/components/GridView.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ export default function GridView(props: ViewPropsType) {
   const { schema, path, data } = props;
   const { properties, view = {} } = schema as ObjectSchemaType;
   const { alignX, alignY, align_x, align_y, gap = 1, orientation } = view;
-  const direction = orientation === "vertical" ? "row" : "column";
+  const direction = orientation === "horizontal" ? "row" : "column";
```
Comment (ritch) [inline app/packages/core/src/plugins/SchemaIO/components/GridView.tsx:12]: ...so this was broken this whole time!?

File: app/packages/core/src/plugins/SchemaIO/components/GridView.tsx [hunk 2]
```diff
@@ -18,14 +18,20 @@ export default function GridView(props: ViewPropsType) {
...
+  const adjustedLayoutWidth = getAdjustedLayoutWidth(
+    props?.layout?.width,
+    parsedGap
+  );
 
   const baseGridProps: BoxProps = {
     sx: {
-      display: "grid",
-      gap,
+      display: "flex",
```
Comment (ritch) [inline app/packages/core/src/plugins/SchemaIO/components/GridView.tsx:29]: big change... how was this tested?

**Review by ritch — APPROVED** [review]

File: app/packages/core/src/plugins/SchemaIO/components/GridView.tsx (hunk 1 above)
Comment (imanjra) [reply app/packages/core/src/plugins/SchemaIO/components/GridView.tsx:12]: It did work but It's reversed for flex interestingly

File: app/packages/core/src/plugins/SchemaIO/components/GridView.tsx (hunk 2 above)
Comment (imanjra) [reply app/packages/core/src/plugins/SchemaIO/components/GridView.tsx:29]: Tested with the existing panels and ensuring they look same as before these changes with better auto-flow

**Pull request merged**